sdks/python/apache_beam/runners/worker/channel_factory.py

```diff
@@ -23,8 +23,24 @@
 
 class GRPCChannelFactory(grpc.StreamStreamClientInterceptor):
   DEFAULT_OPTIONS = [
-      ("grpc.keepalive_time_ms", 20000),
-      ("grpc.keepalive_timeout_ms", 300000),
+      # Default: 30000ms (30s), increased to 180s to reduce ping frequency
+      ("grpc.keepalive_time_ms", 180000),
+      # Default: 5000ms (5s), increased to 10 minutes for stability
+      ("grpc.keepalive_timeout_ms", 600000),
+      # Default: 2, set to 0 to allow unlimited pings without data
+      ("grpc.http2.max_pings_without_data", 0),
+      # Default: False, set to True to allow keepalive pings when no calls
+      ("grpc.keepalive_permit_without_calls", True),
+      # Default: 300000ms (5min), increased to 10min for stability
+      ("grpc.http2.min_recv_ping_interval_without_data_ms", 600000),
+      # Default: 300000ms (5min), increased to 120s for conservative pings
+      ("grpc.http2.min_sent_ping_interval_without_data_ms", 120000),
+      # Default: 2, set to 0 to allow unlimited ping strikes
+      ("grpc.http2.max_ping_strikes", 0),
+      # Default: 0 (disabled), enable socket reuse for better handling
+      ("grpc.so_reuseport", 1),
+      # Default: 0 (disabled), set 30s TCP timeout for connection control
+      ("grpc.tcp_user_timeout_ms", 30000),
```

Comment:
does python use c-core for grpc? if so it seems that this does default to 20s not 0

where are you seeing the defaults?